src/patternfly/components/Select/select.scss

```diff
@@ -144,6 +146,19 @@
   background-color: var(--pf-c-select__toggle--BackgroundColor);
   border: none;
 
+  &.pf-m-disabled,
+  &:disabled {
+    pointer-events: none;
+
+    &:not(.pf-m-plain) {
```

Comment:
you can remove this selector, too. we want the plain variation to get the background color and `border: 0`